alerts/absent_alerts.libsonnet

```diff
@@ -0,0 +1,46 @@
+{
+  prometheusAlerts+:: {
+    groups+: [
+      {
+        name: 'ceph-absent',
+        rules: [
+          {
+            alert: 'CephMgrIsAbsent',
+            expr: |||
+              absent(up{%(cephExporterSelector)s} == 1)
+            ||| % $._config,
+            'for': '5m',
+            labels: {
+              severity: 'warning',
+            },
+            annotations: {
+              message: 'Ceph Manager has disappeared from Prometheus target discovery.',
+              component: 'ceph-manager',
+              grafana_url: '%(grafanaMgrDashboardURL)s' % $._config,
```

Comment:
Can you please explain the use-case for grafana URL in the alert.  
Currently we are following the following format for the annotations.

Reply:
Also I feel its better not having grafana reference from inside prometheus. Let grafana be a layer on top of prometheus which may or may not be present. So by default having grafana reference inside prometheus rules is something which doesnt look good.

Reply:
The use case for grafana url, is simple you click on alert annotation's URL  and it sends you to grafana, where you see it rendered in a graph. IMO it cuts down mean time to resolve as you get to see the state things are in, really fast.

Maybe we should add link to ceph dashboard instead?

Reply:
@devopsjonas I understand the usecase. My point is that instead of doing reverse reference we always should refer prometheus from UI/Grafana. Prometheus alerts in itself are complete and if some component needs to consume details, they should rather refer it.
Makes sense?

Reply:
Not really, as you get alerts in opsgenie/pagerduty/zabbix, you don't get them in grafana. Atleast we don't.

So the typical workflow is `Pagerduty` alert -> click link -> `grafana` / `ceph dashboard`.

Reply:
I've also created an issue for this https://github.com/ceph/ceph-mixins/issues/54 I think the way this https://blog.pvincent.io/2017/12/prometheus-blog-series-part-5-alerting-rules/#provide-context-to-facilitate-resolution blog tells to do it makes sense. And hopefully I can convince you guys to do it :)